src/validation.cpp

```diff
@@ -837,6 +837,10 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
         return state.Invalid(TxValidationResult::TX_INPUTS_NOT_STANDARD, "bad-txns-nonstandard-inputs");
     }
 
+    if (DummyScriptBytes(tx, m_view) > m_pool.m_max_dummy_script_bytes.value_or(0)) {
```

Comment:
For the default case where no valid tx can possibly have more dummy script data than is allowed, evaluating this at all seems pointless. Could gate it on `tx.GetTotalSize() <= m_max_dummy_script_bytes`, perhaps.

Reply:
Maybe I'm missing something but if I replace the condition with `tx.GetTotalSize() <= m_max_dummy_script_bytes` , the `DummyScriptBytes()` function will not be called in the rest of the code, so how can it detect inscriptions (or other arbitrary data)?